### ISSUE TITLE
Using RadeonHD/Warp3D for Southern Islands cards on X5000

### ISSUE THREAD

**Issue by elfpipe** [Sakura-IT/ReWarp3DPPC#5]
I am trying to get Wipeout running on X5000 with RadeonHD 7700. I am using the SI Warp3D driver from AmiStore.

To get 3d graphics working, I have to do a custom build of Warp3DPPC.library (rewarp), that disables the z-buffer (which means just returning W3D_SUCCESS in W3D_AllocZBuffer and W3D_ClearZBuffer).

This of course garbles the graphics, although it runs at a very satisfying speed.

Can you help with ideas how to bring this solution further?

**Comment by DvdBoon** [comment]
Isn't NovaBridge filling in this gap?

**Comment by DvdBoon** [comment]
Besides, warp3dppc.library is just a wrapper to the OS4 warp3d.library.

This is the first WipEout, right?

**Comment by DvdBoon** [comment]
Normal allocating and freeing zbuffer goes through w3d_Picasso96.library (at least with the old Warp3D) with W3DGFX_AllocVMem/FreeVMem.

Clearing is HW specific

**Comment by elfpipe** [comment]
> Isn't NovaBridge filling in this gap?

Yes. I was unable to get it, so I wanted to try alternatives. Very
interesting.

> Besides, warp3dppc.library is just a wrapper to the OS4 warp3d.library.

> This is the first WipEout, right?
Yes. Actually there is one usecase, where it makes a lot of sense.
When using nova, you can't get fullscreen in wipeout. By spoofing 32 as
16 bit in Warp3DPPC.library, you can get it working.

> Normal allocating and freeing zbuffer goes through w3d_Picasso96.library
(at least with the old Warp3D) with W3DGFX_AllocVMem.

I know too little about it to evaluate, what that means. But there is
definitely a problem with the z-buffer when using "classic" Warp3D+SI
driver.
The triangles are not organized right, and there are big areas unrendered.

Den søn. 14. jul. 2024 kl. 22.41 skrev DvdBoon ***@***.***>:
